packages/components/src/DataList/DataList.tsx

```diff
@@ -47,17 +50,8 @@ export function DataList<T extends DataListObject>({
         {title && <Heading level={3}>{title}</Heading>}
         <DataListTotalCount totalCount={totalCount} loading={loading} />
       </div>
-      <div className={styles.header}>
-        {/* Filters here, since it also sticks to the top */}
-        {headerData && layout?.(headerData)}
-      </div>
-      {layout &&
-        elementData.map((child, i) => (
-          // TODO: Don't use index as key. Might have to force an ID on the data JOB-76773
-          <div className={styles.listItem} key={i}>
-            {layout(child)}
-          </div>
-        ))}
+      {headerData && dataListHeaderContent?.map(_layout => _layout)}
```

Comment:
There should only be 1 header, no?

Reply:
Because it renders based on the layout it could change for different layouts instead of just the first layout

Reply:
This may be able to be changed since I took a different approach with handling layouts. It is a small change so look at the mediaquery hook approach and let me know